### PR TITLE
Add .NET language 

### PR DESCRIPTION
**Pull request by AdamTylerLynch**
*Description of changes:*
Added .NET to the repo to clarify support for Graviton and Graviton2 based instances.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by AGSaidi — COMMENTED** [review]
Why mention arm32?

**Comment by AdamTylerLynch** [comment]
> Why mention arm32?

If a team is currently targeting .NET Core 2.1 (which has support through August 2021) they would need to know which instance type to target.

File: dotnet.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# .NET on Graviton
+
+.NET is a free, cross-platform, open source developer platform for building many different types of applications. .NET applications are written in the C#, F#, or Visual Basic programming language. Code is compiled into a language-agnostic Common Intermediate Language (CIL). Compiled code is stored in assemblies—files with a .dll or .exe file extension. For more information, please see [what is .NET](https://dotnet.microsoft.com/learn/dotnet/what-is-dotnet).
```
Comment (AGSaidi) [inline dotnet.md:3]: We can't just copy the text from a different website

File: dotnet.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+[.NET Framework 4.x](https://dotnet.microsoft.com/learn/dotnet/what-is-dotnet-framework) | No | No | The original implementation of the .NET Framework does not support Linux hosts, and Windows hosts are not suported on Graviton. 
+[.NET Core 3.1](https://dotnet.microsoft.com/download/dotnet/3.1) | Yes | Yes | .NET Core 3.0 added support for [Arm64 for Linux](https://docs.microsoft.com/en-us/dotnet/core/whats-new/dotnet-core-3-0#linux-improvements). 
+[.NET Core 2.1](https://dotnet.microsoft.com/download/dotnet/2.1) | Yes* | No | Initial support was for [Arm32 was added to .NET Core 2.1](https://github.com/dotnet/announcements/issues/82). *Operating system support is limited, please see the [official documentation](https://github.com/dotnet/core/blob/main/release-notes/2.1/2.1-supported-os.md).
+
+
+## .NET 5
+The .NET team has significantly improved performance with .NET 5, both generally and for Arm64. The team focused on Arm64-specific optimizations in the .NET libraries and evaluation of code quality produced by RyuJIT and resulting outcomes.
```
Comment (AGSaidi) [inline dotnet.md:19]: looks like this is verbatim from https://devblogs.microsoft.com/dotnet/arm64-performance-in-net-5/ we can't just copy it. please submit content that you've written yourself.

File: dotnet.md (hunk 1 above)
Comment (AdamTylerLynch) [reply dotnet.md]: Rewrote copy.

File: dotnet.md (hunk 2 above)
Comment (AdamTylerLynch) [reply dotnet.md]: Rewrote copy.

File: dotnet.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+## .NET 5
+With .NET 5 Microsoft has made specific Arm64 architecture optimizations. These optimizations were made in the .NET libraries as well as in the machine code output by the JIT process.
+
+ * AWS Compute Blog [Powering .NET 5 with AWS Graviton2: Benchmarks](https://aws.amazon.com/blogs/compute/powering-net-5-with-aws-graviton2-benchmark-results/) 
+ * Microsoft .NET Blog [ARM64 Performance in .NET 5](https://devblogs.microsoft.com/dotnet/arm64-performance-in-net-5/)
+
+
+## Building & Publishing for Linux Arm64
+The .NET SDK supports choosing a [Runtime Identifier (RID)](https://docs.microsoft.com/en-us/dotnet/core/rid-catalog) used to target platforms where the applications run. These RIDs are used by .NET dependencies (NuGet packages) to represent platform-specific resources in NuGet packages. The following values are examples of RIDs: linux-x64, ubuntu.14.04-x64, win7-x64, or osx.10.12-x64. For the NuGet packages with native dependencies, the RID designates on which platforms the package can be restored.
```
Comment (AGSaidi) [inline dotnet.md:24]: can you add a arm64 rid :)

**Review by AGSaidi — APPROVED** [review]

**Pull request merged**